Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoEntityOverrides.cpp

```diff
@@ -0,0 +1,145 @@
+ *
+ */
+
+#include <AzToolsFramework/Prefab/Instance/InstanceToTemplateInterface.h>
+#include <AzToolsFramework/Prefab/Link/Link.h>
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoEntityOverrides.h>
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoUtils.h>
+#include <AzToolsFramework/Prefab/PrefabInstanceUtils.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        PrefabUndoEntityOverrides::PrefabUndoEntityOverrides(const AZStd::string& undoOperationName)
+            : UndoSystem::URSequencePoint(undoOperationName)
+            , m_linkId(InvalidLinkId)
+            , m_changed(true)
+        {
+            m_prefabSystemComponentInterface = AZ::Interface<PrefabSystemComponentInterface>::Get();
+            AZ_Assert(m_prefabSystemComponentInterface, "PrefabUndoEntityOverrides - Failed to grab prefab system component interface.");
+
+            m_instanceToTemplateInterface = AZ::Interface<InstanceToTemplateInterface>::Get();
+            AZ_Assert(m_instanceToTemplateInterface, "PrefabUndoEntityOverrides - Could not get instance to template interface.");
+        }
+
+        bool PrefabUndoEntityOverrides::Changed() const
+        {
+            return m_changed;
+        }
+
+        void PrefabUndoEntityOverrides::Capture(
+            const AZStd::vector<const AZ::Entity*>& entityList,
+            Instance& owningInstance,
+            const Instance& focusedInstance)
+        {
+            InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(owningInstance, &focusedInstance);
+            AZ_Assert(climbUpResult.m_isTargetInstanceReached, "PrefabUndoEntityOverrides::Capture - "
+                "Owning prefab instance should be a descendant of the focused prefab instance.");
+
+            m_linkId = climbUpResult.m_climbedInstances.back()->GetLinkId();
+            AZ_Assert(m_linkId != InvalidLinkId, "PrefabUndoEntityOverrides::Capture - "
+                "Could not get the link id between focused instance and top instance.");
+
+            LinkReference link = m_prefabSystemComponentInterface->FindLink(m_linkId);
+            if (!link.has_value())
+            {
+                AZ_Error("Prefab", false, "PrefabUndoEntityOverrides::Capture - Could not get the link for override editing.");
+                return;
+            }
+
+            PrefabDomReference cachedOwningInstanceDom = owningInstance.GetCachedInstanceDom();
+
+            const AZStd::string overridePatchPathToOwningInstance = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(
+                climbUpResult.m_climbedInstances, true); // skip top instance
+
+            for (const AZ::Entity* entity : entityList)
+            {
+                if (!entity)
+                {
+                    AZ_Error("Prefab", false, "PrefabUndoEntityOverrides::Capture - Found a null entity. Skipping it.");
```

Comment:
warning might be more appropriate here since this won't be a user generated error and the user can't do anything to fix it